rs/crypto/ecdsa_secp256k1/tests/tests.rs

```diff
@@ -128,6 +128,26 @@ fn should_accept_signatures_that_we_generate() {
     }
 }
 
+#[test]
+fn should_accept_bip340_signatures_that_we_generate() {
+    use rand::RngCore;
+
+    let mut rng = test_rng();
+
+    for _ in 0..100 {
+        let sk = PrivateKey::generate_using_rng(&mut rng);
+
+        let pk = sk.public_key();
+        println!("{}", hex::encode(pk.serialize_sec1(true)));
+
+        let mut msg = [0u8; 32];
```

Comment:
We should randomize the message as well. I was going to suggest to use variable size messages, but I am not sure we support this atm.

Reply:
Yeah we're limited here by `k256 0.13`. This is fixed already in `k256` master. I asked in https://github.com/RustCrypto/elliptic-curves/issues/1068 what the plan is for a new release. This is kind of obnoxious, in that in this specific case the state machine tests can't support something that does work on the IC, but it's (IMO) fairly niche, and should be addressed as soon as a new `k256` is out.

Reply:
Good to hear they fixed it. Let's create a ticket to fix this here once k256 has a new version.

Reply:
We can still add the randomization of the messages though.

Reply:
Randomized the test